packages/@sanity/cli/src/actions/init-project/createCliConfig.ts

```diff
@@ -14,33 +14,57 @@ export default defineCliConfig({
    * Enable auto-updates for studios.
    * Learn more at https://www.sanity.io/docs/cli#auto-updates
    */
-  autoUpdates: true,
+  autoUpdates: __BOOL__autoUpdates__,
 })
 `
 
 export interface GenerateCliConfigOptions {
   projectId: string
   dataset: string
+  autoUpdates: boolean
 }
 
 export function createCliConfig(options: GenerateCliConfigOptions): string {
   const variables = options
   const template = defaultTemplate.trimStart()
   const ast = parse(template, {parser})
+
   traverse(ast, {
     StringLiteral: {
       enter({node}) {
         const value = node.value
         if (!value.startsWith('%') || !value.endsWith('%')) {
           return
         }
-
         const variableName = value.slice(1, -1) as keyof GenerateCliConfigOptions
         if (!(variableName in variables)) {
           throw new Error(`Template variable '${value}' not defined`)
         }
-
-        node.value = variables[variableName] || ''
+        const newValue = variables[variableName]
+        /*
+         * although there are valid non-strings in our config,
+         * they're not in StringLiteral nodes, so assume undefined
+         */
+        node.value = typeof newValue === 'string' ? newValue : ''
+      },
+    },
+    Identifier: {
+      enter({node}) {
+        if (!node.name.startsWith('__BOOL__')) {
+          return
+        }
+        const variableName = node.name.replace(
+          /^__BOOL__(.+?)__$/,
+          '$1',
+        ) as keyof GenerateCliConfigOptions
+        if (!(variableName in variables)) {
+          throw new Error(`Template variable '${variableName}' not defined`)
+        }
+        const value = variables[variableName]
+        if (typeof value !== 'boolean') {
+          throw new Error(`Expected boolean value for '${variableName}'`)
+        }
+        node.name = value.toString()
```

Comment:
I don't like this approach - we're relying on the fact that babel won't validate that the name is valid. This identifier node should be replaced with a Literal node.  I'll do a fix.